include/nlohmann/detail/iterators/iteration_proxy.hpp

```diff
@@ -122,25 +149,34 @@ template<typename IteratorType> class iteration_proxy
 {
   private:
     /// the container to iterate
-    typename IteratorType::reference container;
+    typename IteratorType::pointer container;
```

Comment:
Double check if changing this to a pointer has any implications elsewhere in the code. Specifically, I remember an assert that may have a different meaning now. (Also set to `nullptr`.)